tests/test_model_tool.py

```diff
@@ -0,0 +1,114 @@
+from json import loads as json_loads
+from os.path import join
+from unittest import TestCase
+
+from sortedcontainers import SortedSet
+
+from cyclonedx.exception.model import MutuallyExclusivePropertiesException
+from cyclonedx.model.bom import Bom
+from cyclonedx.model.component import Component
+from cyclonedx.model.service import Service
+from cyclonedx.model.tool import Tool, ToolsRepository
+from cyclonedx.output.json import JsonV1Dot5
+from cyclonedx.output.xml import XmlV1Dot5
+from tests import OWN_DATA_DIRECTORY, reorder
+from tests._data.models import get_bom_with_tools_with_component_and_service
+
+
+class TestModelTool(TestCase):
+
+    def test_sort(self) -> None:
+        # expected sort order: (vendor, name, version)
+        expected_order = [0, 1, 2, 3, 4, 5, 6]
+        tools = [
+            Tool(vendor='a', name='a', version='1.0.0'),
+            Tool(vendor='a', name='a', version='2.0.0'),
+            Tool(vendor='a', name='b', version='1.0.0'),
+            Tool(vendor='a', name='b'),
+            Tool(vendor='b', name='a'),
+            Tool(vendor='b', name='b', version='1.0.0'),
+            Tool(name='b'),
+        ]
+        sorted_tools = sorted(tools)
+        expected_tools = reorder(tools, expected_order)
+        self.assertListEqual(sorted_tools, expected_tools)
+
+    def test_tool_with_component_and_service_load_json(self) -> None:
+        test_file = join(OWN_DATA_DIRECTORY, 'json', '1.5',
+                         'bom_with_tool_with_component_and_service.json')
+        with open(test_file, encoding='UTF-8') as f:
+            bom_json = json_loads(f.read())
+        bom = Bom.from_json(bom_json)   # type: ignore[attr-defined]
+        good_bom = get_bom_with_tools_with_component_and_service()
+        self.assertTrue(bom == good_bom)
+
+    def test_tool_with_component_and_service_render_json(self) -> None:
```

Comment:
@jkowalleck Are these "render" tests even needed since I made `get_bom_with_tools_with_component_and_service` and that auto-generates a test and snapshot file?

Reply:
I assume I'd still want the "load" tests.

Reply:
tests might be revisited some day in a far far future ;-)